submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,101 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammal {
+   constructor(species, name, gender, legs, friends, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.friends = friends;
+      this.saying = saying;
+   }
+
+   tellAbout() {
+      const properties = ['species', 'name', 'gender']; //all mammals have this properties
+      for (let property in this) {
+         if (!properties.includes(property))
+            properties.push(property);
+      }
+      return properties.map(property => `<b>${property}:</b> ${this[property]}`).join('\t');
+   }
+};
+
+class Dog extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('dog', name, gender, 4, friends, saying || 'guf');
+   }
+};
+
+class Cat extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('cat', name, gender, 4, friends, saying || 'meow');
+   }
+
+   getSaying() {
+      return this.saying || 'meow';
```

Comment:
Now we have literal `'meow'` as a default value at two locations in the code.
This is an error prone approach. Someone may change it in one place and forget to change it in any other.
Perhaps we need a constant for this. But is there a chance that `this.saying` won't have definite value at this point?
It may happen that we do not need `|| 'meow'` here. Test it.